boards/litex/arty/README.md

```diff
@@ -0,0 +1,136 @@
+- built around a VexRiscv-CPU
+- along with the following configuration options:
+
+  ```
+  --uart-baudrate=1000000
+  --cpu-variant=full
+  --csr-data-width=32
+  --timer-uptime
+  --integrated-rom-size=0xb000
+  --with-ethernet
+  ```
+
+Many bitstream customizations can be represented in the Tock board by
+simply changing the variables in `src/litex_generated.rs`. To support
+a different set of FPGA cores and perform further modifications, the
+`src/main.rs` file will have to be modified.
+
+
+Please note
+-----------
+
+This board is still in development. The memory protection (PMP)
+mechanism is not yet integrated into the VexRiscv core and more
+peripherals require drivers. Nonetheless, the Tock kernel works and
+can run multiple userspace applications.
+
+The following on-board components and cores are supported:
+- [X] Timer (with uptime support)
+- [X] UART output via USB-FTDI
+- [X] Green onboard LEDs
+- [X] 100MBit/s Ethernet MAC
+
+The following components and cores require porting:
+- [ ] Memory protection (PMP) support in the VexRiscv CPU ([upstream
+      PR](https://github.com/SpinalHDL/VexRiscv/pull/147))
+- [ ] GPIO Interface
+- [ ] Buttons and Switches
+- [ ] RGB LEDs
+
+
+Building the SoC / Programming the FPGA
+---------------------------------------
+
+Please refer to the [LiteX
+documentation](https://github.com/enjoy-digital/litex/wiki/) for
+instructions on how to install and use the LiteX SoC generator.
+
+Once LiteX and Xilinx Vivado is installed, building a bitstream should
+be as simple as:
+
+```
+$ cd $PATH_TO_LITEX/litex/boards/targets
+$ ./arty.py <configuration options> --build
+```
+
+This will produce a folder `build/arty/gateware` containing the
+generated bitstream for the FPGA (`arty.bin`).
+
+In addition to that, a folder `build/arty/software` will be included
+containing support code and the SoC bios stored in ROM. The individual
+SoC configuration options, interrupt assignments, register (LiteX
+configuration status registers) addresses, etc. relevant for Tock can
+be found in
+`build/arty/software/include/generated/{csr.h,regions.ld,soc.h}`.
+
+The bitstream can be programmed either by using Xilinx Vivado or running:
+
+```
+$ ./arty.py <configuration options> --load
+```
+
+To persistently write the bitstream to the included SPI flash, use the
+Xilinx Vivado tools as outlined in [this
+manual](https://reference.digilentinc.com/learn/programmable-logic/tutorials/arty-programming-guide/start#programming_the_arty_using_quad_spi). The
+SPI flash used on the board may deviate from the manual, the part
+number to select in Xilinx Vivado will be written on the SPI flash
+chip.
+
+
+Programming
+-----------
+
+By default, the LiteX SoC will feature an integrated BIOS in ROM,
+which acts as a bootloader. The Tock kernel binary can be loaded
+either using `litex_term.py` (available as `lxterm`) via serial, or
+using TFTP via Ethernet. The uploaded image will be placed into the
+`main_ram` section and executed.
```

Comment:
Can you add instructions on how to do this, or add it to the Makefile?

Reply:
Sure. If you're referring to the TFTP-based upload, that's described below. I can add a code snippet for the invocation of `litex_term.py` though.